### PR TITLE
[ie/cbc.ca:player] Support new URL format

### LINKED ISSUE #9534: CBC player has new URL formats that breaks downloads
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Worldwide

### Provide a description that is worded well enough to be understood

The URLs used to use the same ID that was also used on ThePlatform, meaning that the URL could be easily transformed into a URL for ThePlatform (see line 244 in [https://github.com/yt-dlp/yt-dlp/blob/e5d4f11104ce7ea1717a90eea82c0f7d230ea5d5/yt_dlp/extractor/cbc.py#L244](url)). The URL now uses an internal ID instead, which breaks the extractor. I haven't gotten around to fixing the code, but I have had the time to look into what is needed to extract the video:

- The regex needs to be changed to accept a dot in the ID
- The webpage needs to be downloaded and json code needs to be extracted (the same component needs to be extracted as [https://github.com/yt-dlp/yt-dlp/blob/e5d4f11104ce7ea1717a90eea82c0f7d230ea5d5/yt_dlp/extractor/cbc.py#L273](url)
- In the json code, the media ID has to be extracted, it's located in `video/currentClip/mediaID`
- that mediaID can then be used to construct the ThePlatform URL in the same way as before

Maybe I'll get around to taking care of this in the coming days. If anyone else wants to take care of it, feel free.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['https://www.cbc.ca/player/play/1.7154309', '-vU']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8 (No VT), error utf-8 (No VT), screen utf-8 (No VT)
[debug] yt-dlp version stable@2024.03.10 from yt-dlp/yt-dlp [615a84447] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-7-6.1.7601-SP1 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: none
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.02.02, mutagen-1.47.0, requests-2.31.0, sqlite3-3.35.5, urllib3-2.2.1, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Post-Processor Plugins: CheckVideoIntegrityPP
[debug] Plugin directories: ['C:\\Users\\Benjamin\\AppData\\Local\\yt-dlp\\yt-dlp-plugins\\CheckVideoIntegrity\\yt_dlp_plugins', 'C:\\Users\\Benjamin\\AppData\\Local\\yt-dlp\\yt-dlp-plugins\\yt-dont-lock-p\\yt_dlp_plugins']
[debug] Loaded 1803 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.03.10 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.03.10 from yt-dlp/yt-dlp)
[cbc.ca:player] Extracting URL: https://www.cbc.ca/player/play/1.7154309
[ThePlatform] Extracting URL: http://link.theplatform.com/s/ExhSPC/media/guid/2655402169/1?mbr=true&formats=MPEG4,FLV,MP3#__youtubedl_smuggle=%7B%22force_smil_url%22%3A+true%7D
[ThePlatform] 1: Downloading SMIL data
ERROR: [ThePlatform] 1: Unable to download XML: HTTP Error 404: Not Found (caused by <HTTPError 404: Not Found>)
  File "yt_dlp\extractor\common.py", line 732, in extract
  File "yt_dlp\extractor\theplatform.py", line 315, in _real_extract
  File "yt_dlp\extractor\theplatform.py", line 36, in _extract_theplatform_smil
  File "yt_dlp\extractor\common.py", line 1086, in download_content
  File "yt_dlp\extractor\common.py", line 1050, in download_handle
  File "yt_dlp\extractor\adobepass.py", line 1366, in _download_webpage_handle
  File "yt_dlp\extractor\common.py", line 920, in _download_webpage_handle
  File "yt_dlp\extractor\common.py", line 877, in _request_webpage

  File "yt_dlp\extractor\common.py", line 864, in _request_webpage
  File "yt_dlp\YoutubeDL.py", line 4101, in urlopen
  File "yt_dlp\networking\common.py", line 115, in send
  File "yt_dlp\networking\_helper.py", line 204, in wrapper
  File "yt_dlp\networking\common.py", line 326, in send
  File "yt_dlp\networking\_requests.py", line 351, in _send
yt_dlp.networking.exceptions.HTTPError: HTTP Error 404: Not Found
```

### PR DESCRIPTION
**Pull request by trainman261**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

This fixes the CBCPlayerIE after a change in the URL structure broke it. The URL no longer holds the video ID from ThePlatform, but an internal ID instead. As a result, the IE needs to extract the "mediaID" from the webpage. After that has finished, everything else can resume as usual. ~As to the question of which ID should be used by yt-dlp internally - I've stuck with the mediaID, but am unsure of whether that's the best decision. Essentially it comes down to:~
~Pros of the current approach:~

- ~This allows one ID to be used across the CBCPlayerIE, CBCCaffeinePlayerIE (see below) and ThePlatformIE~
- ~This doesn't break the archival of previous videos using the `--download-archive` option, and this option would make it recognize links pointing to the same video regardless of if it's a link handled by CBCPlayerIE or CBCCaffeinePlayerIE (again, see below)~

~Cons of the current approach/pros of using the new ID:~

- ~When checking a list of URLs (i.e. extracted from a playlist or similar), the previous archival status of the links could be checked without extracting without downloading anything from the web page.~

~The direct Caffeine Player links still have the mediaID though, so I've split off the CBCCaffeinePlayerIE from the CBCPlayerIE, which basically has the same code as the old CBCPlayerID (extract mediaID from url, pass it on to ThePlatformIE). (Also, I'm unsure if the IE name is ideal - if anyone has any better suggestions, please let me know.)
In addition, I've slightly changed the regex from the CBCIE which was capturing direct links to the CBC Caffeine Player, leading to an extraction failure. I'm not sure if this was a problem previously, but it isn't now (anymore).~
I've also updated the tests, including changing the `creator` field to the `creators` field.
~Finally, I've slightly changed the regexes to match the request in `CONTRIBUTING.md` to not capture groups that aren't used. Only this change is contained in the last commit, if I understood it wrong, please feel free to remove.~
EDIT: changed it back to being only one IE. The IE now looks for a period in the ID and decides accordingly if it should hand that ID to the ThePlatformIE or extract the MediaID from the page as per @bashonly 's suggestion. In case it is needed in the future due to ID changes, I still have the original code on my computer which could then be recommitted.

Fixes #9534

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [X] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [ ] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [X] Fix or improvement to an extractor (Make sure to add/update tests)
- [X] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by trainman261** [comment]
Not sure why the Core test is failing. It fails at `test_no_duplicates`, which my gut tells me should be from the `_VALID_URL` regex in the `CBCIE`, but I specifically changed that so that there are _no_ duplications - previously, that was the case, even entirely before this PR. Unless I messed it up - but in that case, I can't find where the error is.

**Comment by bashonly** [comment]
why add another extractor? is there ID overlap? can't we tweak the url regex and just only request the webpage if the `media_id` is not matched in the `_VALID_URL`?

**Comment by bashonly** [comment]
ah oops I misspoke, I meant `CBCPlayerIE`, not `CBCIE`

```py
    _VALID_URL = r'(?:cbcplayer:|https?://(?:www\.)?cbc\.ca/(?:player/play/|i/caffeine/syndicate/\?mediaId=))(?P<id>(?:\d\.)?\d+)'
```

and then

```py
        video_id = self._match_id(url)
        if '.' in video_id:
            webpage = self._download_webpage(url, video_id)
            json_content = self._search_json(
                r'window\.__INITIAL_STATE__\s*=', webpage, 'initial state', video_id)
            video_id = json_content['video']['currentClip']['mediaId']
```

**Comment by trainman261** [comment]
...and I misread. My bad.
My reason for adding the second IE was that judging by the way the video player works, matching the URL format should be more reliable than matching the ID format in the long term. If the ID format changes, it just needs a slight regex change.
On the other hand, for the time being, this solution is a lot more elegant. I'll switch it back, and if we ever need it the other way around, this code can always be pulled back.

**Review by bashonly — APPROVED** [review]

File: yt_dlp/extractor/cbc.py [hunk 1]
```diff
@@ -165,9 +165,52 @@ class CBCPlayerIE(InfoExtractor):
...
         },
         'skip': 'Geo-restricted to Canada and no longer available',
+    }, {
+        'url': 'http://www.cbc.ca/i/caffeine/syndicate/?mediaId=2657631896',
+        'md5': 'e5e708c34ae6fca156aafe17c43e8b75',
+        'info_dict': {
+            'id': '2657631896',
+            'ext': 'mp3',
+            'title': 'CBC Montreal is organizing its first ever community hackathon!',
+            'description': 'The modern technology we tend to depend on so heavily, is never without it\'s share of hiccups and headaches. Next weekend - CBC Montreal will be getting members of the public for its first Hackathon.',
```
Comment (bashonly) [inline yt_dlp/extractor/cbc.py:175]:
```suggestion
            'title': 'CBC Montreal is organizing its first ever community hackathon!',
            'description': 'md5:dd3b692f0a139b0369943150bd1c46a9',
```

**Pull request merged**